Formula/gismo.rb

```diff
@@ -0,0 +1,68 @@
+class Gismo < Formula
+  desc "C++ library for isogeometric analysis (IGA)"
+  homepage "https://gismo.github.io"
+  url "https://github.com/gismo/gismo/archive/refs/tags/v21.12.0.tar.gz"
+  sha256 "4001b4c49661ca8b71baf915e773341e115d154077bef218433a3c1d72ee4f0c"
+  license "MPL-2.0"
+
+  head do
+    url "https://github.com/gismo/gismo.git", branch: "stable"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "openblas"
+  depends_on "suite-sparse"
+  depends_on "superlu"
+
+  on_macos do
+    depends_on "libomp"
+  end
+
+  def install
+    ENV.runtime_cpu_detection
+
+    args = std_cmake_args
+    args << "-DBLA_VENDOR=OpenBLAS"
+    args << "-DGISMO_WITH_SUPERLU=ON"
+    args << "-DGISMO_WITH_UMFPACK=ON"
+
+    if OS.mac?
+      libomp = Formula["libomp"]
+      args << "-DOpenMP_C_FLAGS=-Xpreprocessor -fopenmp -I#{libomp.opt_include}"
+      args << "-DOpenMP_C_LIB_NAMES=omp"
+      args << "-DOpenMP_CXX_FLAGS=-Xpreprocessor -fopenmp -I#{libomp.opt_include}"
+      args << "-DOpenMP_CXX_LIB_NAMES=omp"
+      args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.a"
+    end
+    args << "-DGISMO_WITH_OPENMP=ON"
+
+    if OS.mac?
+      args << "-DTARGET_ARCHITECTURE=apple-m1" if Hardware::CPU.arm?
+      args << "-DTARGET_ARCHITECTURE=penryn"   if Hardware::CPU.intel?
+    elsif OS.linux?
+      args << "-DTARGET_ARCHITECTURE=penryn"   if Hardware::CPU.intel?
+    end
```

Comment:
Why does it not work with Homebrew's detection of this?

Reply:
I tried using Homebrew's detection mechanism before but the naming convention in our CMake script is not identical to what is returned by Homebrew's detection mechanism. That is, I need the strings `apple-m1` and `penryn`. If you prefer, I can revert to Homebrew's detection mechanism and do something like this
```
args << case Hardware.oldest_cpu
when :arm_vortex_tempest
  "-DTARGET_ARCHITECTURE=apple-m1"
else
  "-DTARGET_ARCHITECTURE=penryn"
end
```

In an earlier version I had
```
else
  "-DTARGET_ARCHITECTURE=" << Hardware.oldest_cpu.upcase.to_s
end
```
but the returned `core2` is not supported by our CMake script.

Reply:
And what if you pass nothing?

Reply:
Then our CMake script determines the OS/ARCH combination of the machine the build process is running on and optimizes the binaries and libraries for that particular machine. As you raise this issue, is there a way to detect if the formula is built locally (`--build-from-source`) or as a bottle. In the former case it might be beneficial to build with optimized compiler flags while bottles should be built to run on all supported CPUs.

Reply:
Okay, I'll do that in the next commit.